operator/handler/config.go

```diff
@@ -95,35 +104,34 @@ func combineConfigs(annotation string, sidecarSpecs map[string]tailingsidecarv1.
 				},
 			}
 			configs = append(configs, config)
-		case configPredefined:
-			if _, ok := sidecarSpecs[configParts[containerNameIndex]]; !ok {
-				handlerLog.Info("Missing configuration in TailingSidecarConfig configurations",
-					"configuration name", configParts[containerNameIndex],
-				)
-				continue
-			}
-			config := sidecarConfig{
-				name: configParts[containerNameIndex],
-				spec: sidecarSpecs[configParts[containerNameIndex]],
-			}
-			configs = append(configs, config)
 		default:
-			handlerLog.Info("Incorrect configuration",
+			handlerLog.Info("Incorrect format of 'tailing-sidecar' annotation",
 				"annotation", annotation)
 		}
 	}
 	return configs
 }
 
-// joinTailingSidecarSpecs joins configurations defined in TailingSidecarConfig resources
-func joinTailingSidecarSpecs(tailingSidecars []tailingsidecarv1.TailingSidecarConfig) map[string]tailingsidecarv1.SidecarSpec {
-	sidecarSpecs := make(map[string]tailingsidecarv1.SidecarSpec, len(tailingSidecars))
+// convertTailingSidecarConfigs converts configurations defined in TailingSidecarConfigs to sidecarConfig
+func convertTailingSidecarConfigs(tailingSidecars []tailingsidecarv1.TailingSidecarConfig) ([]sidecarConfig, error) {
+	sidecarNames := make(map[string]interface{}, len(tailingSidecars))
```

Comment:
Not that big of a difference but it makes sure that there's no meaning behind the values stored.

```suggestion
	sidecarNames := make(map[string]struct{}, len(tailingSidecars))
```